plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -194,51 +189,58 @@ private CollectedStatistics mergeStatisticsIfNecessary(
                 updateMode = REPLACE;
             }
         }
+        Map<Integer, CompactSketch> collectedNdvSketches = collectedStatistics.ndvSketches();
+        StatsUpdateMode finalUpdateMode = updateMode;
+        Set<Integer> columnsWithRecentlyComputedStats = new HashSet<>(collectedNdvSketches.keySet());
 
-        return switch (updateMode) {
-            case REPLACE -> collectedStatistics;
-            case INCREMENTAL_UPDATE -> {
-                Map<Integer, CompactSketch> collectedNdvSketches = collectedStatistics.ndvSketches();
-                ImmutableMap.Builder<Integer, CompactSketch> ndvSketches = ImmutableMap.builder();
-
-                Set<Integer> pendingPreviousNdvSketches = new HashSet<>(collectedNdvSketches.keySet());
-                Iterator<StatisticsFile> statisticsFiles = walkStatisticsFiles(table, snapshotId);
-                while (!pendingPreviousNdvSketches.isEmpty() && statisticsFiles.hasNext()) {
-                    StatisticsFile statisticsFile = statisticsFiles.next();
-
-                    boolean hasUsefulData = statisticsFile.blobMetadata().stream()
-                            .filter(blobMetadata -> blobMetadata.type().equals(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1))
-                            .filter(blobMetadata -> blobMetadata.fields().size() == 1)
-                            .anyMatch(blobMetadata -> pendingPreviousNdvSketches.contains(getOnlyElement(blobMetadata.fields())));
+        Optional<StatisticsFile> latestStatisticsFile = getLatestStatisticsFile(table, snapshotId);
+        ImmutableMap.Builder<Integer, CompactSketch> ndvSketches = ImmutableMap.builder();
+        if (latestStatisticsFile.isEmpty()) {
+            if (updateMode == REPLACE) {
+                return collectedNdvSketches;
```

Comment:
Why do this only if no previous statistics file? Shouldn't REPLACE mean always replace?